src/main/java/org/mqttbee/mqtt/datatypes/MqttUTF8StringImpl.java

```diff
@@ -68,32 +68,51 @@
 
     /**
      * Validates and decodes a UTF-8 encoded String from the given byte array.
+     * <p>
+     * Note: the given byte array must not be longer than {@link MqttBinaryData#MAX_LENGTH}.
      *
      * @param binary the byte array with the UTF-8 encoded data to decode from.
      * @return the created UTF-8 encoded String or null if the byte array does not contain a well-formed UTF-8 encoded
      * String.
      */
     @Nullable
     public static MqttUTF8StringImpl from(@NotNull final byte[] binary) {
-        return containsMustNotCharacters(binary) ? null : new MqttUTF8StringImpl(binary);
+        return binary.length > MqttBinaryData.MAX_LENGTH || containsMustNotCharacters(binary) ? null :
```

Comment:
please wrap in parenthesis: `((... > ...) || ...)`

Reply:
DONE